### ISSUE TITLE
[GG] Reuse MLA context accumulators in place for DS4F and GLM-5.2

### ISSUE THREAD

**Issue by voipmonitor** [local-inference-lab/vllm#470]
## Purpose

Status: research-only.

Evaluate in-place accumulation of chunked MLA context results for DeepSeek-V4/DS4F and GLM-5.2 on `dev/gilded-gnosis`.

`vllm/model_executor/layers/attention/mla_attention.py` currently uses two attention-output tensors and two LSE tensors as a ping-pong accumulator. The source and destination never alias, but the second BF16 output tensor and FP32 LSE tensor remain allocated while context chunks are merged.

The proposed implementation uses one attention-output tensor and one LSE tensor as both the running prefix and merge destination. The native CUDA kernel must first satisfy that aliasing contract. [vLLM PR #468](https://github.com/local-inference-lab/vllm/pull/468) implements and tests the required `merge_attn_states` behavior.

## Applicability

DeepSeek-V4/DS4F and GLM-5.2 use the shared MLA chunked-context accumulation path when their selected attention backend returns multiple context partials. The optimization affects chunked prefill and prefix restoration; it does not change decode execution.

For an accumulator containing `T` query tokens, `H` local heads, and value width `D`, the expected per-rank transient-memory reduction is:

```text
T * H * D * sizeof(attention dtype) + T * H * sizeof(float LSE)
```

The actual reduction and throughput effect must be measured for each production profile.

## Required implementation

1. Port the single-allocation output and LSE accumulator to `dev/gilded-gnosis` without changing the MLA backend interface.
2. Include PR #468 or an equivalent fix before permitting `output_lse` to alias `prefix_lse`.
3. Preserve non-contiguous head strides, partial-token ranges, DCP context partitioning, and CUDA Graph capture behavior.
4. Retain the disjoint-buffer implementation as the reference oracle during qualification. A permanent runtime fallback is not required if the in-place contract is fully qualified.

## Qualification

- Compare aliased and disjoint merges exactly for every DS4F and GLM-5.2 head geometry at scheduler chunks of 256 and 4096 tokens.
- Exercise the first context partial, continuation partials, empty-token ranges, `-inf` LSE values, DCP partitions, and CUDA Graph replay.
- Run paired DS4F and GLM-5.2 E2E captures with identical stored token IDs for cold chunked prefill and restored-prefix prefill.
- Verify logits or pre-LM-head states against the disjoint-buffer reference. The optimization is acceptable only when measured divergence remains at the deterministic repeat floor.
- Record peak allocated and reserved VRAM, accumulator allocation sizes, and 8k/32k/64k prefill throughput with at least three paired repetitions.
- Confirm that no-spec and speculative decode throughput remains within measurement noise because the changed path is prefill-only.

## Completion criteria

The optimization is qualified when both model profiles demonstrate the expected transient-memory reduction, all aliasing tests pass, E2E numerical comparison shows no additional drift, and prefill throughput has no statistically supported regression.
